common/http-api-client/src/lib.rs

```diff
@@ -803,6 +803,57 @@ impl ApiClientCore for Client {
     }
 }
 
+/// An axum test client useful for writing unit tests
+#[cfg(feature = "axum_test")]
+pub struct AxumTestClient {
+    axum_test: axum_test::TestServer,
+}
```

Comment:
my suggestion is to make the client take the refecence to the TestServer so that after you send your request, you would still have access to the server instance in your tests to make additional assertions on

```suggestion
pub struct AxumTestClient<'a> {
    axum_test: &'a axum_test::TestServer,
}
```